hera_cal/lst_stack/tests/test_binning.py

```diff
@@ -404,29 +407,164 @@ def test_redavg_with_where_inpainted(self, request, tmp_path_factory):
             binning.lst_bin_files_from_config(cfg)
 
 
-class TestLSTStack:
-    def setup_class(self):
-        self.uvd = mockuvd.create_mock_hera_obs(
-            integration_time=1.0, ntimes=20, jd_start=2459844.0, ants=[0, 1, 2, 3],
-            time_axis_faster_than_bls=False
-        )
-        self.stack = binning.LSTStack(self.uvd)
+# Silence the expected antpos warning from the binning code
+pytestmark = pytest.mark.filterwarnings(
+    "ignore:Getting antpos from the first file only"
+)
+
+
+class _FakeConfiguratorSingle:
+    """Minimal interface that SingleBaselineStacker uses."""
+    def __init__(self, bl_to_file_map):
+        self.bl_to_file_map = bl_to_file_map
+
+
+# --- fixtures using real test files -------------------------------------------------
+
+@pytest.fixture(scope="module")
+def real_files_and_grid():
+    baseline_string = "0_4"
+    filenames = [
+        "zen.2459861.baseline.0_4.sum.smooth_calibrated.red_avg.inpainted.uvh5",
+        "zen.2459862.baseline.0_4.sum.smooth_calibrated.red_avg.inpainted.uvh5",
+        "zen.2459863.baseline.0_4.sum.smooth_calibrated.red_avg.inpainted.uvh5",
```

Comment:
While it's a good idea to test the reading of real (downsampled) files, the problem is that in one year, we've forgotten what's inside these files, so the tests below that use these files for things like testing if the LST wrap worked are harder to know whether they're doing what we expect, because we don't know the properties of the input data. 

Using the framework I laid out above would allow you to also construct instances directly from small test arrays that you can define in each function, making it more obvious what the test is testing for.